iup/__init__.py

```diff
@@ -831,3 +846,446 @@ def predict(
+    def validate(self):
+        self.assert_type_included(pl.Date)
+        self.assert_type_included(pl.Float64)
+        # has at least 1 column of date and 1 column of estimate
+
+    def assert_type_included(self, datatype):
+        """
+        Verify at least one column has the expected type.
+
+        Parameters
+        columns: List[str]
+            names of columns for which to check type
+        datatype:
+            data type for each listed column
+        """
+        assert any(
+            dtype == datatype for dtype in self.schema.values()
+        ), f"No column is {datatype} type"
+
+
+class QuantileForecast(Forecast):
+    """
+    Class for forecast with quantiles.
+    Save for future.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.validate()
+
+    # Must be named as "quantileX.XX" except the date column
+    def validate(self):
+        estimate = self.select(pl.all().exclude(pl.Date))
+        QuantileForecast.assert_column_name_all(estimate, "quantile")
+
+    @staticmethod
+    def assert_column_name_all(estimate, column_name):
+        """
+        Verify that all columns have a pattern with a common name
+
+        Parameters
+        estimate:
+            The data without 'date' type
+        column_name:
+            The common column name
+        """
+        pattern = rf"^{column_name}\d(\.\d+)?$"
+        assert all(
+            [bool(re.match(pattern, col)) for col in estimate.columns]
+        ), f"Not all columns are Column name {column_name}"
+
+
+class PointForecast(QuantileForecast):
+    """
+    Class for forecast with point estimate
+    A subclass when quantile is 0.5
+    For now, enforce the "quantile0.5" to be "estimate"
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.validate()
+
+    def validate(self):
+        self.assert_columns_found(["estimate"])
+        self.assert_columns_type(["estimate"], pl.Float64)
+
+    def assert_columns_found(self, columns: List[str]):
+        """
+        Verify that expected columns are found.
+
+        Parameters
+        columns: List[str]
+            names of expected columns
+        """
+        assert isinstance(columns, List), f"{columns} must be a list"
+
+        for col in columns:
+            assert col in self.columns, f"Column {col} is expected but not found."
+
+    def assert_columns_type(self, columns: List[str], dtype):
+        """
+        Verify that columns have the expected type.
+
+        Parameters
+        columns: List[str]
+            names of columns for which to check type
+        dtype:
+            data type for each listed column
+        """
+        assert isinstance(columns, List), f"{columns} must be a list"
+
+        for col in columns:
+            assert (
+                self[col].dtype == dtype
+            ), f"Column {col} should be {dtype} but is {self[col].dtype}"
+
```

Comment:
`assert_columns_found` and `assert_columns_type` are already methods of UptakeData. Instead of repeating these methods here, maybe we should just make them methods of the parent class common to UptakeData and PointForecast.

Reply:
Actually I thought about this, but it seems like the methods can only be directly-inherited, i.e, the methods can be passed to its direct subclass only. Or it will be concise to put these two methods in `ValidateData`.

Reply:
Yeah, I think this is another facet of the consolidation theme that runs through most of my comments. We'll decide on a broader strategy first, which will hopefully dictate what we do in this particular case.